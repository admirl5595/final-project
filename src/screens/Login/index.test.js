import React from "react";
import renderer from "react-test-renderer";

// functions for simulating user interaction
import { render, fireEvent } from "@testing-library/react-native";

// import component to be tested
import LoginScreen from "./index";

// mock hooks from react native navigation
jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
    useRoute: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
      params: { patient: 1 },
    }),
  };
});

// mock hooks from react native navigation
jest.mock("../../../firebase-config", () => {
  const actualConfig = jest.requireActual("../../../firebase-config");
  return {
    ...actualConfig,
    auth: { currentUser: { displayName: "bob" } },
  };
});

describe("check contents", () => {
  it("has email input field", () => {
    const { getByPlaceholderText, getByText, getAllByText } = render(
      <LoginScreen />
    );

    const emailInput = getByPlaceholderText("Email");

    expect(emailInput).toBeTruthy();
  });

  it("has password input field", () => {
    const { getByPlaceholderText, getByText, getAllByText } = render(
      <LoginScreen />
    );

    const passwordInput = getByPlaceholderText("Password");

    expect(passwordInput).toBeTruthy();
  });
});
