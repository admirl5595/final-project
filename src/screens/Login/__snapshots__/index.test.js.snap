// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check contents shows header and icon 1`] = `[Function]`;
